projects/tasks.py

```diff
@@ -127,3 +130,587 @@ def handle_too_many_daypass_users(artifact_uuid):
         message=strip_tags(body),
         html_message=body,
     )
+
+
+def get_project_link(project):
+    return f"https://chameleoncloud.org/user/projects/{project.pk}"
+
+
+def email_subject_1_project(project):
+    return f"Your project {project.charge_code} has been automatically tagged!"
+
+
+def email_subject_n_projects(n):
+    return f"{n} of your projects have been automatically tagged!"
+
+
+BASE_INTRO = """
+            <p>
+            As part of a recent change in our project organization, we have implemented 
+            a new project tagging system.
+"""
+
+
+def email_body_intro_1_project(project, runners_up):
+    body = (
+        BASE_INTRO
+        + f"""
+             We automatically selected the following tag
+             for your project <b>{project.charge_code} — {project.title}</b>:
+             
+            <p><i>{project.tag.name} — {project.tag.description}</i></p>
+            
+            </p>
+            """
+    )
+    if runners_up:
+        body += f"""
+            <p>The following tags were considered, but not chosen:</p>
+                
+            <p>{runners_up}</p>
+        """
+
+    body += f"""
+    <p>
+    If our selection does not seem correct to you, you may freely update your
+    tag <a href="{get_project_link(project)}" target="_blank">here</a>.
+    </p>
+    """
+
+    return body
+
+
+def email_body_intro_n_projects(projects):
+    project_list = "\n<br>".join(
+        f'<li> <b><a href="{get_project_link(p)}" target="_blank">{p.charge_code}</a> '
+        f"— {p.description}</b>\n<ul><li>{p.tag.name} — {p.tag.description}</ul>"
+        for p in projects
+    )
+    return (
+        BASE_INTRO
+        + f"""
+    We automatically selected new tags for your projects:<br><br>
+    <ol>
+    {project_list}
+    </ol>
+    </p>
+    
+    <p>
+    If our selections do not seem correct to you, you may freely update your tags by
+    visiting the links to your projects above.
+    </p>
+    """
+    )
+
+
+def automatically_tag_projects():
```

Comment:
To run this manually, you should be able to `exec` into the django shell, and them `from project.tasks import automatically_tag_projects; automatically_tag_projects()`.